File: mysql_examples/recipes/api/04_stmt/placeholder.py
#!/usr/bin/python
# placeholder.py - demonstrate statement processing in Python
# (with placeholders)

import Cookbook
import MySQLdb

try:
  conn = Cookbook.connect ()
except MySQLdb.Error, e:
  print "Cannot connect to server"
  print "Error code:", e.args[0]
  print "Error message:", e.args[1]
  sys.exit (1)

print "Execute INSERT statement using placeholders"
try:
#@ _EXECUTE_PLACEHOLDER_1_
  cursor = conn.cursor ()
  cursor.execute ("""
                  INSERT INTO profile (name,birth,color,foods,cats)
                  VALUES(%s,%s,%s,%s,%s)
                  """, ("De'Mont", "1973-01-12", None, "eggroll", 4))
#@ _EXECUTE_PLACEHOLDER_1_
  print "Number of rows inserted: %d" % cursor.rowcount
except MySQLdb.Error, e:
  print "Oops, the statement failed"
  print e

print "Execute SELECT statement using placeholders"
try:
#@ _EXECUTE_PLACEHOLDER_2_
  cursor = conn.cursor ()
  cursor.execute ("SELECT * FROM profile WHERE cats = %s", (2,))
  for row in cursor.fetchall ():
    print row
  cursor.close ()
#@ _EXECUTE_PLACEHOLDER_2_
except MySQLdb.Error, e:
  print "Oops, the statement failed"
  print e

print "Construct INSERT statement using literal()"
try:
#@ _LITERAL_
  cursor = conn.cursor ()
  stmt = """
         INSERT INTO profile (name,birth,color,foods,cats)
         VALUES(%s,%s,%s,%s,%s)
         """ % \
         (conn.literal ("De'Mont"), \
         conn.literal ("1973-01-12"), \
         conn.literal (None), \
         conn.literal ("eggroll"), \
         conn.literal (4))
  cursor.execute (stmt)
#@ _LITERAL_
  print "Statement:"
  print stmt
  print "Number of rows inserted: %d" % cursor.rowcount
except:
  print "Oops, the statement failed"

conn.close ()

File: mysql_examples/recipes/api/04_stmt/placeholder.php
<?php
# placeholder.php - demonstrate statement processing in PHP
# (with placeholders)

require_once "Cookbook.php";

$conn =& Cookbook::connect ();
if (PEAR::isError ($conn))
{
  print ("Cannot connect to server.\n");
  printf ("Error message: %s\n", $conn->getMessage ());
  exit (1);
}

# Pass data values directly to query()

print ("Execute INSERT statement with do()\n");
#@ _PLACEHOLDER_QUERY_
$result =& $conn->query ("INSERT INTO profile (name,birth,color,foods,cats)
                          VALUES(?,?,?,?,?)",
                         array ("De'Mont","1973-01-12",NULL,"eggroll",4));
if (PEAR::isError ($result))
  die ("Oops, the statement failed");
#@ _PLACEHOLDER_QUERY_
printf ("Number of rows inserted: %d\n", $conn->affectedRows ());

# Prepare a statement, then pass the statement and data values to execute()

print ("Execute INSERT statement with prepare() + execute()\n");
#@ _PLACEHOLDER_PREPARE_EXECUTE_1_
$stmt =& $conn->prepare ("INSERT INTO profile (name,birth,color,foods,cats)
                          VALUES(?,?,?,?,?)");
if (PEAR::isError ($stmt))
  die ("Oops, statement preparation failed");
$result =& $conn->execute ($stmt,
                           array ("De'Mont","1973-01-12",NULL,"eggroll",4));
if (PEAR::isError ($result))
  die ("Oops, statement execution failed");
#@ _PLACEHOLDER_PREPARE_EXECUTE_1_
printf ("Number of rows inserted: %d\n", $conn->affectedRows ());

print ("Select rows using placeholder\n");
#@ _PLACEHOLDER_PREPARE_EXECUTE_2_
$result =& $conn->query ("SELECT * FROM profile WHERE cats > ?", array (2));
if (PEAR::isError ($result))
  die ("Oops, the statement failed");
while ($row =& $result->fetchRow (DB_FETCHMODE_ASSOC))
{
  printf ("id: %s, name: %s, cats: %s\n",
      $row["id"], $row["name"], $row["cats"]);
}
$result->free ();
#@ _PLACEHOLDER_PREPARE_EXECUTE_2_

print ("Construct INSERT statement usinq quoteSmart()\n");
#@ _QUOTESMART_
$stmt = sprintf ("INSERT INTO profile (name,birth,color,foods,cats)
                  VALUES(%s,%s,%s,%s,%s)",
                 $conn->quoteSmart ("De'Mont"),
                 $conn->quoteSmart ("1973-01-12"),
                 $conn->quoteSmart (NULL),
                 $conn->quoteSmart ("eggroll"),
                 $conn->quoteSmart (4));
$result =& $conn->query ($stmt);
if (PEAR::isError ($result))
  die ("Oops, the statement failed");
#@ _QUOTESMART_
print ("Statement:\n$stmt\n");
printf ("Number of rows inserted: %d\n", $conn->affectedRows ());

$conn->disconnect ();
?>

